woocommerce/woocommerce-gateway-stripe
# Miscellaneous tasks on ECE

Handled some miscellaneous tasks in this PR.

## Changes and testing instructions
- Passed `locale` to ECE when initializing ([c2e3a6f](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3551/commits/c2e3a6fbc1ff28c77f40bf153f61aecb4f056811))
- Replaced instances of `source` with `payment method` on the edit subscription page ([c6f730b](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3551/commits/c6f730b040ede139c3146701ab8536b42af115c8))
    - Go to the subscription edit page and click the pencil icon in the billing address section.
    - Before the payment method field used to be `Stripe Sources ID`. Now it is changed to `Stripe Payment Method ID`

<img width="591" alt="Screenshot 2024-10-30 at 3 29 00 PM" src="https://github.com/user-attachments/assets/82a7737c-9e37-4548-b5ca-e0b8e4941836">

- Not calling the ECE specific `paymentFailed` function for order creation failures ([071cd30](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3551/commits/071cd3083370a2bad032cfa4ac1c65e4e5819130))
    - Fixed the `abortPayment` function. When `abortPayment` is called due to order related failure (not payment failure with ECE), the `paymentFailed( { reason: 'fail' } )` call throws an error.

<img width="431" alt="Screenshot 2024-10-28 at 4 14 57 PM" src="https://github.com/user-attachments/assets/3cd47b8f-bcfd-4ee4-85f2-e5cd812ea299">


- Fixed issue in purchasing subscriptions when the store has no shipping options ([148b75b](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3551/commits/148b75b1e00bdbbcefeba2e76df6397b460e39cb))

    - Remove all the shipping methods from your store.
    - Create a simple subscription product.
    - As a shopper, go the the simple subscription product page.
    - Try to pay with Google Pay.
    - In `develop`, you should see the above error, but in this branch payment should work fine.

<img width="1019" alt="Screenshot 2024-10-28 at 11 37 12 AM" src="https://github.com/user-attachments/assets/09e355be-1d55-4f94-99dc-6c1d87ce49a8">
